cmd/agent/main.go

```diff
@@ -365,6 +384,13 @@ func run(ctx context.Context) error {
 		if !isConnect() {
 			wsURL += "/register"
 		}
+
+		// check if we need a new kubelet cert on reconnection
+		err = KubeletNeedsNewCertificate(headers)
```

Comment:
I just realized something here -- this will only evaluate the certificate on disk for expiration on connection -- if the certificate is getting close to expiration (long lived agent), will it properly re-generate a new certificate to replace the old  one?

I'm not even sure if the original agent would have done that FWIW, but we should probably fix that. I do worry that swapping the certificate out from under the kubelet might also cause the kubelet to freak out/potentially cause downtime...

Reply:
That is a good point. I didn’t see any code relating to proactive refreshing of certs when they’re close to expiring, even on the rancher side. I can’t speak for who implemented it initially, but maybe the expiry time of ten years played a part in that decision. Should we try to handle that case here, or create a new issue for it? 

Plus the agent has been wildly swapping out certificates with the kubelet running so far, so maybe that’s another behavior we want to look at.

Reply:
You're right, it looks for a new one every two minutes. I've added a check which ensures the expiry date isn't within three days right before we request the plan, this should handle the long lived connection issue. I'm not sure if we should discuss additional logic to handle the swapping of the cert, so far it looks like swapping it in this way hasn't caused issues for the kubelet.